### PR TITLE
Reworked simulations renderers

### PR DESCRIPTION
**Pull request by favreau**
…g speed

### CODE REVIEW

File: plugins/engines/ospray/ispc/render/ParticleRenderer.ispc [hunk 1]
```diff
@@ -1,9 +1,7 @@
-/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+/* Copyright (c) 2015-2017, EPFL/Blue Brain Project
```
Comment (karjonas) [inline plugins/engines/ospray/ispc/render/ParticleRenderer.ispc]: 2018

File: plugins/engines/ospray/ispc/render/ParticleRenderer.ispc [hunk 2]
```diff
@@ -20,154 +18,142 @@
...
     int64 simulationDataSize;
-    uniform vec4f* uniform transferFunction;
-    uniform float* uniform transferFunctionEmissionData;
-    uint32 transferFunctionSize;
+
+    // Alpha correction
+    float alphaCorrection;
 };
 
 inline varying vec4f
```
Comment (karjonas) [inline plugins/engines/ospray/ispc/render/ParticleRenderer.ispc]: A similar function is used also for the simulation renderer, can they be merged?

**Comment by favreau** [comment]
Retest this please

File: plugins/engines/ospray/ispc/render/utils/SimulationRenderer.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#include <ospray/SDK/common/Material.h>
+#include <ospray/SDK/render/Renderer.h>
+
+// system
+#include <vector>
+
+namespace brayns
+{
+/**
+ * The AbstractRenderer class implements a base renderer for all Brayns custom
```
Comment (karjonas) [inline plugins/engines/ospray/ispc/render/utils/SimulationRenderer.h]: Fix documentation.

File: plugins/engines/ospray/ispc/render/utils/SimulationRenderer.ih [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const uint64 index = (uint64)(dg->st.x * OFFSET_MAGIC) << 32 |
+                         (uint32)(dg->st.y * OFFSET_MAGIC);
+
+    if (index < self->simulationDataSize)
+        value = self->simulationData[index];
+    else
+        // Value offset is out of range, return error color
+        return color;
+
+    // Normalize the value according colormap size
```
Comment (karjonas) [inline plugins/engines/ospray/ispc/render/utils/SimulationRenderer.ih]: according (to) colormap

File: plugins/engines/ospray/ispc/render/BasicSimulationRenderer.ispc [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <plugins/engines/ospray/ispc/render/utils/SimulationRenderer.ih>
+
+struct BasicSimulationRenderer
+{
+    SimulationRenderer abstract;
```
Comment (karjonas) [inline plugins/engines/ospray/ispc/render/BasicSimulationRenderer.ispc]: Is there a better name than abstract, like simulation_renderer or something?

File: plugins/engines/ospray/ispc/render/AdvancedSimulationRenderer.ispc [hunk 6]
```diff
@@ -21,46 +21,35 @@
...
+#include <plugins/engines/ospray/ispc/render/utils/SimulationRenderer.ih>
 
-// needs to be the same in MorphologyLoader.cpp
-uniform const uniform float OFFSET_MAGIC = 1e6;
-
-struct SimulationRenderer
+struct AdvancedSimulationRenderer
 {
-    AbstractRenderer abstract;
+    SimulationRenderer abstract;
```
Comment (karjonas) [inline plugins/engines/ospray/ispc/render/AdvancedSimulationRenderer.ispc]: Is there a better name than abstract, like simulation_renderer or something?

**Comment by ppodhajski** [comment]
retest this please

File: plugins/engines/ospray/ispc/render/BasicSimulationRenderer.ispc (hunk 5 above)
Comment (favreau) [reply plugins/engines/ospray/ispc/render/BasicSimulationRenderer.ispc]: Renamed to super (to remain consistent with OSPRay)

File: plugins/engines/ospray/ispc/render/AdvancedSimulationRenderer.ispc (hunk 6 above)
Comment (favreau) [reply plugins/engines/ospray/ispc/render/AdvancedSimulationRenderer.ispc]: Dito

File: plugins/engines/ospray/ispc/render/utils/SimulationRenderer.ih (hunk 4 above)
Comment (favreau) [reply plugins/engines/ospray/ispc/render/utils/SimulationRenderer.ih]: Done

File: plugins/engines/ospray/ispc/render/utils/SimulationRenderer.h (hunk 3 above)
Comment (favreau) [reply plugins/engines/ospray/ispc/render/utils/SimulationRenderer.h]: Done

**Comment by karjonas** [comment]
Why is  tests/files/images/testdataallmini50color.png updated? Should it not be the same still?

**Pull request merged**